### PR TITLE
Minor refactor to SourcePartitionedScheduler

### PR DESCRIPTION
**Pull request by wenleix**

### CODE REVIEW

**Comment by wenleix** [comment]
@shixuan-fan : I often find myself needs to reason where does `scheduleGroup.pendingSplits` get copied out. Make always using `scheduleGroup.pendingSplits` seems easier to tracking who change it.

File: presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java [hunk 1]
```diff
@@ -437,14 +437,15 @@ public synchronized void notifyAllLifespansFinishedExecution()
                 .addAll(noMoreSplitsNotification.keySet())
                 .build();
         for (InternalNode node : nodes) {
-            // source partitioned tasks can only receive broadcast data; otherwise it would have a different distribution
             ImmutableMultimap<PlanNodeId, Split> splits = ImmutableMultimap.<PlanNodeId, Split>builder()
                     .putAll(partitionedNode, splitAssignment.get(node))
                     .build();
+
```
Comment (shixuan-fan) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java]: Yet these seem to be unrelated changes ;)

File: presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java [hunk 2]
```diff
@@ -437,14 +437,15 @@ public synchronized void notifyAllLifespansFinishedExecution()
...
             ImmutableMultimap.Builder<PlanNodeId, Lifespan> noMoreSplits = ImmutableMultimap.builder();
             if (noMoreSplitsNotification.containsKey(node)) {
                 noMoreSplits.putAll(partitionedNode, noMoreSplitsNotification.get(node));
             }
+
```
Comment (shixuan-fan) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java]: ditto

**Comment by shixuan-fan** [comment]
Also checkstyle failure in travis is related.

File: presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java (hunk 1 above)
Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java]: Yeah. I found without empty lines it's hard to read sometimes...Squash them into the first commit (it's not real "refactor" but i don't want to make commit title too verbose, like "minor refactor and style adjustment...")  and mention it in the commit message :)

File: presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java [hunk 3]
```diff
@@ -199,15 +199,14 @@ public synchronized ScheduleResult schedule()
...
 
             if (scheduleGroup.state == ScheduleGroupState.NO_MORE_SPLITS || scheduleGroup.state == ScheduleGroupState.DONE) {
                 verify(scheduleGroup.nextSplitBatchFuture == null);
             }
-            else if (pendingSplits.isEmpty()) {
+            else if (scheduleGroup.pendingSplits.isEmpty()) {
                 // try to get the next batch
                 if (scheduleGroup.nextSplitBatchFuture == null) {
-                    scheduleGroup.nextSplitBatchFuture = splitSource.getNextBatch(scheduleGroup.partitionHandle, lifespan, splitBatchSize - pendingSplits.size());
+                    scheduleGroup.nextSplitBatchFuture = splitSource.getNextBatch(scheduleGroup.partitionHandle, lifespan, splitBatchSize);
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java]: 😄
